From 40cdaea4d7824cc1b0d26e6ad2dcb61fa2077911 Mon Sep 17 00:00:00 2001
From: Pino Toscano <pino@kde.org>
Date: Tue, 29 Oct 2019 07:52:32 +0100
Subject: [PATCH] PgSQL driver: fix build with PostgreSQL 12+

ABSTIMEOID and RELTIMEOID were removed, as their data types were dropped.
---
 src/drivers/postgresql/PostgresqlTypes.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git src/drivers/postgresql/PostgresqlTypes.cpp src/drivers/postgresql/PostgresqlTypes.cpp
index ea576d69..0697129e 100644
--- a/src/drivers/postgresql/PostgresqlTypes.cpp
+++ b/src/drivers/postgresql/PostgresqlTypes.cpp
@@ -36,6 +36,7 @@
 #endif
 #include <libpq-fe.h>
 #include <catalog/pg_type.h> // needed for BOOLOID, etc.
+#include <pg_config.h> // needed for PG_VERSION_NUM
 
 #ifdef _MSC_VER
 #pragma warning( pop )
@@ -70,8 +71,10 @@ void PostgresqlDriver::initPgsqlToKDbMap()
     //! @todo POLYGONOID geometric polygon '(pt1,...)'
     m_pgsqlToKDbTypes.insert(FLOAT4OID, KDbField::Double);
     m_pgsqlToKDbTypes.insert(FLOAT8OID, KDbField::Double);
+#if PG_VERSION_NUM < 120000
     m_pgsqlToKDbTypes.insert(ABSTIMEOID, KDbField::Date);
     m_pgsqlToKDbTypes.insert(RELTIMEOID, KDbField::Date);
+#endif
     //! @todo TINTERVALOID (abstime,abstime), time interval
     //! @todo CIRCLEOID geometric circle '(center,radius)'
     //! @todo CASHOID monetary amounts, $d,ddd.cc
-- 
GitLab

